crates/solidity/inputs/language/src/definition.rs

```diff
@@ -5553,6 +5555,22 @@ codegen_language_macros::compile!(Language(
                             identifier = YulIdentifier,
                             definitions = [KeywordDefinition(value = Atom("iszero"))]
                         ),
+                        Keyword(
```

Comment:
Q: do we also need to add them as `BuiltInFunction` entries below?

Reply:
cc @ggiraldez

Reply:
I guess it depends on what's the decision on how to deal with Yul built-ins. If we remove all the keywords and bind them as built-ins, then for sure. If they remain as keywords, then it's not necessary.

In any case, I'd like a way to bind built-ins only inside assembly blocks. It wouldn't be correct to declare them as regular `BuiltInFunction`s because those would bind in regular Solidity code. Maybe we can declare a type `YulBuiltIns` and introduce a hard-coded binding rule that links this scope only when inside an `assembly` statement.

Reply:
Sorry if these are dumb questions, but how is that different from the current `YulBuiltInFunction` type? Also, shouldn't the entire Yul* types be scoped within the `assembly` statement?

Regarding this PR, we have two options:
 1. We merge it as is (correcting the version bump), it being consistent with how Yul built-ins were treated so far.
 2. We create a new PR placing Yul built-ins _somewhere_ (see questions above), and then re-create this PR on top of it.

Reply:
>but how is that different from the current YulBuiltInFunction type?

`YulBuiltInFunction` is a parser kind, but `BuiltInFunction` is used at the end of the file to generate a Solidity source file with declaration of built-in functions available for name binding.

We're analyzing removing the Yul reserved keywords used for built-ins and declare them as built-ins for binding only. So they would stop being recognized as keywords. This will make the grammar simpler.

>Also, shouldn't the entire Yul* types be scoped within the assembly statement?

In the context of the parser, yes they are. For bindings we don't have a mechanism to do that yet. Although it's easy to do, it creates another semi-hidden dependency between the binding rules and the language definition and built-ins generation.

Reply:
>Regarding this PR, we have two options:

I propose we move ahead with 1. for now.

I think it's a good idea to go in the direction of removing Yul built-ins from the grammar, but that's a much bigger change and there are still some unknowns. It also changes the public API in a non-backwards compatible way. Not sure if there are users depending on the existing grammar kinds.

Reply:
> Maybe we can declare a type YulBuiltIns and introduce a hard-coded binding rule that links this scope only when inside an assembly statement.

> it creates another semi-hidden dependency between the binding rules and the language definition and built-ins generation.
 
@ggiraldez that is a good idea. We already have a similar design for grammar using `LexicalContext`. For example, adding a `lexical_context: Option<Identifier>` field to `BuiltIn` types in the DSL. It would also work if we remove `YulBuiltInFunction`, and the dependency above would be explicit.

> It also changes the public API in a non-backwards compatible way. Not sure if there are users depending on the existing grammar kinds.

It would, and we would have very limited chance to do these kinds of changes after we release v1.

> I propose we move ahead with 1. for now.

Sounds good to me, as long the new addition would behave similar to the existing ones in bindings.